paddle/fluid/pir/dialect/operator/ir/ops.yaml

```
@@ -1194,6 +1194,15 @@
   backward : prod_grad
   interfaces : paddle::dialect::InferSymbolicShapeInterface
 
+- op : prune_gate_by_capacity
+  args : (Tensor gate_idx, Tensor expert_count, int64_t n_expert, int64_t n_worker)
+  output : Tensor(new_gate_idx)
+  infer_meta :
+    func : PruneGateByCapacityInferMeta
+  kernel :
+    func : prune_gate_by_capacity
+    data_type : dtype
```

```suggestion
    data_type : gate_idx
```
这个字段是指kernel的dtype要和哪个输入保持一致

---

已修改~